src/main/java/com/example/Game.java

```diff
@@ -80,8 +73,12 @@ public Game() {
         }
 
         char cpuSymbol = (playerSymbol == 'X') ? 'O' : 'X';
-        MoveStrategy strategy = (difficultyLevel == 1) ? new RandomMoveStrategy() : new SmartMoveStrategy();
-        player2 = new CpuPlayer(strategy,cpuSymbol);
+        MoveStrategy strategy = (difficultyLevel == 1)
```

Comment:
Unnecessary complexity by using Combination of factory and strategy,
you are defying the whole purpose of using strategy.

Reply:
- [Article](https://stackoverflow.com/questions/27333278/is-it-okay-to-implement-multiple-design-patterns)

Reply:
I understand about creating complexity in Game class by implementing both Strategy and Factory. But in some cases it's good to use such combinations. Let me explain you what I thought:
Case 1: Factory DP for creating Players
If I have the need to extend my code in future for adding a new Player type in future say RemotePlayer for multiPlayer game. Then I don't have the need to change my logic in Game class I can directly add a new method in Factory class that handles creation of RemotePlayer. So in short it is following Open/Closed solid principle i.e. the game logic in Game class is closed formodification while PlayerFactory is open for modification/extensions if needed.
Case 2: Strategy DP for selecting strategies
Same logic over here no need to modify game class logic while adding new MoveStrategies. Just need to create a strategy write logic for that and if passed to Cpuplayer class for object creation.